src/main/scala/com/rbmhtechnology/calliope/scaladsl/TransactionalEventProducer.scala

```diff
@@ -148,34 +150,82 @@ class TransactionalEventProducer[A] private(sourceId: String, topic: String, eve
-    new EventStoreWriter[A](topic, eventStore, runner ! NotifyCommit)
+  private val graphRunner = system.actorOf(Props(new GraphRunnerSupervisor(graph, onFailure)))
+
+  /**
+    * Creates an [[EventWriter]] which is used to store events to the underlying [[EventStore]] and consecutively produce them to the configured
+    * Kafka endpoint.
+    */
+  def eventWriter(): EventWriter[A] =
+    new EventStoreWriter[A](topic, eventStore, graphRunner ! NotifyCommit)
+
+  /**
+    * Starts producing events from the underlying [[EventStore]] to the configured Kafka endpoint.
+    *
+    * This is an asynchronous operation which returns a [[Future]] that is resolved once event production has started or fails if the given
+    * timeout was exceeded.
+    */
+  def run()(implicit timeout: FiniteDuration = 5.seconds): Future[Done] =
+    askGraphRunnerTo(RunGraph)
+
+  /**
+    * Stops producing events from the underlying [[EventStore]].
+    *
+    * This is an asynchronous operation which returns a [[Future]] that is resolved once event production has stopped or fails if the given
+    * timeout was exceeded.
+    *
+    * The TransactionalEventProducer may be started afterwards by invoking `start()` to continue event production.
+    *
+    * <p>
+    * <b>IMPORTANT:</b>
+    * This method will stop event-production without removing already produced events from the underlying store. Duplicates may be produced
+    * by a subsequent call to `run()` as a consequence.
+    * </p>
+    */
+  def stop()(implicit timeout: FiniteDuration = 5.seconds): Future[Done] =
+    askGraphRunnerTo(StopGraph)
+
+  /**
+    * Terminates the TransactionalEventProducer. No event production can be performed after invoking this method.
+    * Should be used for cleanup purposes.
+    *
+    * This is an asynchronous operation which returns a [[Future]] that is resolved once the component has been terminated or fails if the given
+    * timeout was exceeded.
+    *
+    * <p>
+    * <b>IMPORTANT:</b>
+    * This method will stop event-production without removing already produced events from the underlying store. Duplicates may be produced
+    * by a subsequent incarnation of the [[TransactionalEventProducer]] as a consequence.
+    * </p>
+    */
+  def terminate()(implicit timeout: FiniteDuration = 5.seconds): Future[Done] = {
+    val deadline = Deadline.now + timeout
+    stop().flatMap(_ => gracefulStop(graphRunner, deadline.timeLeft)).map(_ => Done)
   }
 
-  def stop(timeout: FiniteDuration = 5.seconds): Future[Boolean] = {
-    if (running)
-      gracefulStop(graphRunner, timeout)
-    else
-      Future.successful(true)
+  private def askGraphRunnerTo(msg: Any)(implicit timeout: FiniteDuration): Future[Done] = {
+    ask(graphRunner, msg)(Timeout(timeout))
+      .map(_ => Done)
+      .transform(identity, {
+        case _: AskTimeoutException => new RuntimeException(s"TransactionalEventProducer did not respond within the given timeout of $timeout.")
```

Comment:
Why is a `RuntimeException` better than a `AskTimeoutException`? To hide that ask is involved? Maybe we should document that instead, so it is clear how the future fails, if the given timeout is exceeded?!?